### DISCUSSION TITLE
Folders in SharePoint Online Site Page library.

### DISCUSSION THREAD

**Discussion by Troid2k** [WikiTransformationProject/wikitraccs-releases#98] [Q&A]
Does anyone know if there is a way to migrate Confluence Spaces directly into a folder contained in SharePoint Online Site Page library?
To explain further,
I create a target site in SharePoint to migrate Confluence Spaces into. 
I start configure WikiTraccs with all of the correct information to migrate Confluence Spaces into the SharePoint site.
The migrated pages end up in the root of the SharePoint Site Pages Page Library.
But.....
In SharePoint, Pages, I can click the settings cog > Library settings > More library settings > Advanced Settings > Make "New Folder" command Available? > Yes
With that enabled I can create new folders in SharePoint Site Pages Page Library and then copy migrated pages into their own folder.
However....
If I do that I break links in pages.
Has anyone found a way to make this work?

**Comment by heinrich-ulbricht** [comment]
WikiTraccs migrates pages so they are consistent with SharePoint's out of the box pages. That means: pages flat to the Site Pages library, attachments to a corresponding folder in the Site Assets library.

The attachments part is important when it comes to permissions, as SharePoint automatically adjusts folder permissions when changing page permissions.

What problem are you trying to solve?

**Reply by Troid2k** [reply]
Hello Heinrich,
I am trying to organise a friendly SharePoint home page. We have services and each service has it's own Confluence Space. So once migrated I wanted the home page to link through to a folder that contains the pages migrated from their site.
S1, S2, etc... relate to the services.
![SharePoint Home Page](https://github.com/WikiTransformationProject/wikitraccs-releases/assets/30868887/e3e3908b-5732-46c9-a845-c761b342ad7b)
In the Site Pages area I have folders that contain Confluence migrated pages that relate to the confluence space.
![SharePoint Site Pages Folders](https://github.com/WikiTransformationProject/wikitraccs-releases/assets/30868887/4be6ee01-0f82-4d3f-92dd-00d2473a97c3)
Then in the folder we have the pages.
![SharePoint Site Pages](https://github.com/WikiTransformationProject/wikitraccs-releases/assets/30868887/88ebb0a5-3ee8-4c8b-9c0f-b827c7d92e47)

**Reply by Troid2k** [reply]
If I copy the pages into the folder it breaks any links to other pages as they have now copied up a folder level. I was wondering if there would be a way to migrate directly into a folder in the pages area, which might help preserve links as they are transformed. Could it be a new column in the WikiTraccs migration site, Or a feature in the .json config?

**Reply by heinrich-ulbricht** [reply]
@Troid2k Oh that is a tough one. My gut feeling is that your folders should be sites, instead. So each service would migrate into a different target SharePoint site. You can configure this mapping in the Space Inventory. Then you might create a hub site, or other aggregational site, that links to all those service sites. Would that be an option?

To not get lost in migrated pages I created [WikiPakk](https://www.wikipakk.com), the SharePoint page tree. So this could further help the users find their contents, even in a site that contains all pages flat in the Site Pages library, as the spaces would be displayed as separate roots in the page tree.

From a technical standpoint I am hesitant to put pages in folders. I did not see that in large-scale action, yet. And I am suspicious about hidden assumptions baked into the Site Pages library that cause problems down the road. E.g. are those pages compatible with translations? Translated pages also go into folders. Each page has an associated folder in the Site Assets library - does this association still work? Does attaching attachments still work? Synchronized permissions between page and associated folder? I would first have to further investigate this. And only after making sure there would be no problems I could starting to change the algorithm for link generation.

So I don't think migrating pages to folders will be a feature of WikiTraccs in the short-term future. Nevertheless, I'll start searching my network for information about pages and folder compatibility. For now I'd recommend finding an alternative to folders (as long as it is not subsites... they are not recommended by Microsoft and WikiTraccs doesn't migrate to subsites), like sites and hub sites.

Hope this helps somewhat.

**Reply by Troid2k** [reply]
That helps me a lot. Thanks for sharing your ideas.

**Comment by Troid2k** [comment]
Hello Heinrich,

I was trying to make a friendly look to the SharePoint home page by creating buttons on there that link to the folder in the site pages area. That way all of the migrated pages from each confluence space would be in their own folder and they can manage them from there.
I am only able to copy migrated pages into the folders and not move them, which seems to break links in pages. I am able to edit the links with the extra folder level, but I thought if I could edit the migration in some way to fit with the extra folder level as a target it might work. But I can't see a way to make that possible. I will try and add images to better explain tomorrow.

Kind regards

Kevin Partridge
________________________________

**Comment by jlriverayuso** [comment]
wouldnt it be better for Wikitraccs to have a way to update the site navigation based on the Confluence Space Structure? i am actually looking for a way to do this automagically. would it be possible?

**Reply by heinrich-ulbricht** [reply]
@jlriverayuso WikiTraccs doesn't migrate to folders but all pages flat to the Site Pages library. You can migrate multiple spaces to one site, though.

Given the above - how would the site navigation in SharePoint look in your eyes? (Are you thinking navigation within a site, or are you thinking about hub sites?)

Also, what constitutes the "Confluence Space Structure"? Aren't all spaces flat, without hierarchy (at least without any 3rd party addons)?

I'm trying to understand if there's something WikiTraccs could do here. Maybe you have a couple of screenshots, like from Confluence (showing the spaces) and from SharePoint (manually configured, how you expect it to look)?
